azure/scope/clients.go

```diff
@@ -169,3 +205,151 @@ func setValue(settings auth.EnvironmentSettings, key string) {
 		settings.Values[key] = v
 	}
 }
+
+func (c *AzureClients) reconcileASOSecret(ctx context.Context, asoClientSecret *ASOSecretClient) error {
```

Comment:
This function seems big enough that I can see value in adding at least one unit test exercising the happy paths for each of `asoClientSecret.AzureClusterIdentity != nil` and the inverse.